### PR TITLE
consolidated stock move - step 1

### PR DESCRIPTION
**Pull request by VictorEkstrom**

### CODE REVIEW

File: stock_move_consolidated/security/ir.model.access.csv [hunk 1]
```diff
@@ -0,0 +1,2 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_stock_move_consolidation,access_stock_move_consolidation,model_stock_move_consolidation,base.group_user,1,1,1,1
```
Comment (VictorEkstrom) [inline stock_move_consolidated/security/ir.model.access.csv]: We need to be careful that this doesnt make it to a production server, it gives everyone a right to consolidate stock moves.

File: stock_move_consolidated/wizard/stock_move_history.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            items.append({
+                'product_id': rec.product_id.id,
+                'location_id': rec.location_id.id,
+                'location_dest_id': rec.location_dest_id.id,
+                'product_uom_qty': rec.product_uom_qty,
+                'product_uom': rec.product_uom.id,
+            })
+
+        move_dict = OrderedDict()
+        for items in items:
```
Comment (VictorEkstrom) [inline stock_move_consolidated/wizard/stock_move_history.py]: change to for item in items, calling the temporary variable the same name as the list name feels icky

File: stock_move_consolidated/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+{
+    'name': "stock_location_priority",
```
Comment (VictorEkstrom) [inline stock_move_consolidated/__manifest__.py]: you need to give your module its own name and description ;)

File: stock_move_consolidated/models/stock_move.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    #         ('product_id', '=', self.product_id.id),
+    #         '|',
+    #         ('location_id', '=', self.location_id.id),
+    #         ('location_dest_id', '=', self.location_id.id),
+    #         ('lot_id', '=', self.lot_id.id),
+    #         '|',
+    #         ('package_id', '=', self.package_id.id),
+    #         ('result_package_id', '=', self.package_id.id),
+    #     ]
+    #     return action
```
Comment (VictorEkstrom) [inline stock_move_consolidated/models/stock_move.py]: remove all the commented-out stuff that isnt needed (not just here, but everywhere)

File: stock_move_consolidated/wizard/stock_move_history.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            move_dict.setdefault(
+                (
+                    items['product_id'],
+                    items['location_id'],
+                    items['location_dest_id'],
+                    items['product_uom'],
+                ),
+                []).append(items['product_uom_qty'])
```
Comment (VictorEkstrom) [inline stock_move_consolidated/wizard/stock_move_history.py:43]: clever

File: stock_move_consolidated/wizard/stock_move_history.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+                'location_dest_id': k[2],
+                'product_uom': k[3],
+                'product_uom_qty': sum(v)
+            }
+            for k, v in move_dict.items()
+        ]
+
+        for move in move_dict_items:
+            move['name'] = 'Stock Move from %s - %s' % (self.date_from, self.date_to)
+            self.env['stock.move'].create(move)
```
Comment (VictorEkstrom) [inline stock_move_consolidated/wizard/stock_move_history.py:59]: these need to get state "done", or they wont count.

We also need to delete the old stock moves that these replaced, or we wont get the correct amount in storage.

File: stock_move_consolidated/wizard/stock_move_history_view.xml [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+                <footer>
+                    <button name="stock_move_till_date" string="View" type="object" class="btn-primary"/>
+                    <button string="Cancel" class="btn-secondary" special="cancel" />
+                </footer>
+            </form>
+        </field>
+    </record>
+
+    <record id="consolidate_stock_move_action" model="ir.actions.act_window">
```
Comment (VictorEkstrom) [inline stock_move_consolidated/wizard/stock_move_history_view.xml]: we need access to this action without having to mark any products.

Right now i only get the action button after having marked some products, whereas we want this action to simply be based on date

**Review by VictorEkstrom — COMMENTED** [review]
I added a couple of comments that id like you to address miracle, other than that it looks great ^^

File: stock_move_consolidated/security/ir.model.access.csv (hunk 1 above)
Comment (VictorEkstrom) [reply stock_move_consolidated/security/ir.model.access.csv]: the group that gets access to this should be the people who are in the inventory administrator group

File: stock_move_consolidated/__manifest__.py (hunk 3 above)
Comment (iammiracle) [reply stock_move_consolidated/__manifest__.py]: Ah Ah, my bad.

File: stock_move_consolidated/wizard/stock_move_history.py (hunk 2 above)
Comment (iammiracle) [reply stock_move_consolidated/wizard/stock_move_history.py]: I understand, no problem

File: stock_move_consolidated/security/ir.model.access.csv (hunk 1 above)
Comment (iammiracle) [reply stock_move_consolidated/security/ir.model.access.csv]: Yeah, that make sense

File: stock_move_consolidated/wizard/stock_move_history.py (hunk 6 above)
Comment (iammiracle) [reply stock_move_consolidated/wizard/stock_move_history.py:59]: Yes, I will do that, I just want to be sure, we have the correct process

File: stock_move_consolidated/wizard/stock_move_history_view.xml (hunk 7 above)
Comment (iammiracle) [reply stock_move_consolidated/wizard/stock_move_history_view.xml]: okay, then the button on the list view will work. I just added both, but I will remove it

**Comment by VictorEkstrom** [comment]
This code needs to stay on the development branch until we find a way to delete the old stock moves.

Until then, we will just use the odoo8 module and migrating the consolidated moves for the in-going balance

**Pull request closed without merging**